### ISSUE TITLE
jvm crash

### ISSUE THREAD

**Issue by henridf** [joeferner/node-java#284]
The following reliably crashes:
```
var java = require('java');
var Arrays = java.import("java.util.Arrays");
var arr1 = java.newArray("char", "hello world\n".split(''));
var list = Arrays.asList(c);
var arr2 = list.toArray();
```

with the following:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000000107f20b07, pid=93867, tid=2571
#
# JRE version: Java(TM) SE Runtime Environment (8.0_25-b17) (build 1.8.0_25-b17)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.25-b02 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# V  [libjvm.dylib+0x2f9b07]
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/henridf/work/apache-spark-node/hs_err_pid93867.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.sun.com/bugreport/crash.jsp
#
Abort trap: 6
```

**Comment by harmic** [comment]
I had that problem initially (or one that looked very much like it).

In my case it was due to building against one JDK version, and then running against another (I had 1.6 and 1.8 both installed).

I solved it by:

    export JAVA_HOME=/usr/lib/jvm/java-1.8.0
    npm rebuild java

and also making sure that I have JAVA_HOME set to point out the java version you want to use.

I don't make any claim that is the best solution, and in fact it would be better if node-java somehow located the correct version that it was built against.
